airbyte-cdk/python/airbyte_cdk/sources/declarative/decoders/csv_decoder.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+import csv
+import io
+import logging
+from dataclasses import InitVar, dataclass
+from typing import Any, Generator, Mapping, MutableMapping
+
+import requests
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+
+logger = logging.getLogger("airbyte")
+
+
+@dataclass
+class CsvDecoder(Decoder):
+    """
+    CsvDecoder is a decoder strategy that parses the CSV content of the response, and converts it to a stream of dicts.
+    """
+
+    parameters: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, parameters: Mapping[str, Any]):
```

Comment:
Is there a benefit to injecting the delimiter as a parameter property vs directly injecting as an argument?

Reply:
I don't know! Give me a suggestion here, I have no idea how this is done or if it will actually work when instantiated from an actual manifest. It could easily be that this is not the right way.